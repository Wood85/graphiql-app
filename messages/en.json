{
  "Welcome": {
    "title": "Welcome",
    "signIn": "Sign In",
    "signUp": "Sign Up"
  },
  "Header": {
    "home": "Home",
    "signOut": "Sign Out",
    "signIn": "Sign In",
    "signUp": "Sign Up",
    "en": "En",
    "ru": "Ru"
  },
  "About": {
    "title": "About us",
    "description": "This application is a result of our work as a team while studying at RS School, React 2024 Q3 course. We created a light-weight versions of Postman and GraphiQL, combined in one app.",
    "aleksandr": "Aleksandr Krivoshein",
    "ekaterina": "Ekaterina Kotliarenko",
    "mikalai": "Mikalai Shcharbakou",
    "teamLead": "Team-lead",
    "developer": "Frontend developer"
  },
  "SignIn": {
    "title": "Sign In",
    "email": "E-mail",
    "password": "Password",
    "submit": "Submit",
    "description": "Don't have an account",
    "signUp": "Sign Up",
    "emailRequired": "Enter your email",
    "invalidEmail": "Invalid email",
    "enterPassword": "Enter your password",
    "passwordLength": "Password must be at least {passwordLength} symbol",
    "passwordNumber": "Password must contain at least one number",
    "passwordUppercase": "Must contain at least one uppercase letter",
    "passwordLowercase": "Must contain at least one lowercase letter",
    "passwordSpecialChar": "Must contain at least one special character",
    "invalidCredential": "Invalid credential"
  },
  "SignUp": {
    "title": "Sign Up",
    "name": "Name",
    "email": "E-mail",
    "password": "Password",
    "confirmPassword": "Confirm password",
    "submit": "Submit",
    "description": "Already have an account",
    "signIn": "Sign In",
    "nameUppercase": "The first letter must be uppercased",
    "nameRequired": "Enter your name",
    "emailRequired": "Enter your email",
    "invalidEmail": "Invalid email",
    "enterPassword": "Enter your password",
    "passwordLength": "Password must be at least {passwordLength} symbol",
    "passwordNumber": "Password must contain at least one number",
    "passwordUppercase": "Must contain at least one uppercase letter",
    "passwordLowercase": "Must contain at least one lowercase letter",
    "passwordSpecialChar": "Must contain at least one special character",
    "passwordMatch": "Password must match",
    "passwordRequired": "Confirm password is required",
    "userAlreadyExist": "A user with this email already exists",
    "userWasNotCreated": "Error! User was not created"
  }
}
