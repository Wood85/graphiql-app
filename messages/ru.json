{
  "Welcome": {
    "title": "Добро пожаловать",
    "signIn": "Вход",
    "signUp": "Регистрация",
    "restClient": "REST клиент",
    "graphiqlClient": "GraphiQL клиент",
    "history": "История"
  },
  "Header": {
    "home": "Главная",
    "signOut": "Выход",
    "signIn": "Вход",
    "signUp": "Регистрация",
    "en": "Англ",
    "ru": "Рус"
  },
  "About": {
    "title": "О нас",
    "description": "Это приложение - результат нашей совместной работы во время обучения в RS School на курсе React 2024 Q3. Мы создали облегченные версии Postman и GraphiQL, объединенные в одном приложении.",
    "aleksandr": "Александр Кривошеин",
    "ekaterina": "Екатерина Котляренко",
    "mikalai": "Николай Щербаков",
    "teamLead": "Лидер группы",
    "developer": "Фронтенд разработчик"
  },
  "SignIn": {
    "title": "Вход",
    "email": "Почта",
    "password": "Пароль",
    "submit": "Отправить",
    "description": "Пока что нет аккаунта",
    "signUp": "Зарегистрироваться",
    "developer": "Frontend developer",
    "emailRequired": "Введите свой адрес эл. почты",
    "invalidEmail": "Неверный адрес эл. почты",
    "enterPassword": "Введите ваш пароль",
    "passwordLength": "Пароль должен быть не менее {passwordLength} символов",
    "passwordNumber": "Пароль должен содержать хотя бы одну цифру",
    "passwordUppercase": "Должна содержать хотя бы одну заглавную букву",
    "passwordLowercase": "Должна содержать хотя бы одну строчную букву",
    "passwordSpecialChar": "Должен содержать хотя бы один специальный символ",
    "invalidCredential": "Недействительная учетная запись"
  },
  "SignUp": {
    "title": "Регистрация",
    "name": "Имя",
    "email": "Почта",
    "password": "Пароль",
    "confirmPassword": "Подтвердите пароль",
    "submit": "Отправить",
    "description": "Уже есть аккаунт",
    "signIn": "Войти",
    "nameUppercase": "Первая буква должна быть заглавной",
    "nameRequired": "Введите свое имя",
    "emailRequired": "Введите свой адрес эл. почты",
    "invalidEmail": "Неверный адрес эл. почты",
    "enterPassword": "Введите ваш пароль",
    "passwordLength": "Пароль должен быть не менее {passwordLength} символов",
    "passwordNumber": "Пароль должен содержать хотя бы одну цифру",
    "passwordUppercase": "Должна содержать хотя бы одну заглавную букву",
    "passwordLowercase": "Должна содержать хотя бы одну строчную букву",
    "passwordSpecialChar": "Должен содержать хотя бы один специальный символ",
    "passwordMatch": "Пароль должен совпадать",
    "passwordRequired": "Требуется подтверждение пароля",
    "userAlreadyExist": "Пользователь с таким электронным адресом уже существует",
    "userWasNotCreated": "Ошибка! Пользователь не был создан"
  },
  "Restapi": {
    "title": "RESTful Клиент",
    "endpointURL": "URL конечной точки",
    "send": "Отправить",
    "key": "Ключ",
    "value": "Значение",
    "headers": "Заголовки",
    "body": "Тело",
    "variables": "Переменные",
    "text": "Текст",
    "variable": "Переменная",
    "currentValue": "Текущее значение"
  },
  "Response": {
    "body": "Тело",
    "headers": "Заголовки",
    "pretty": "Формат.",
    "raw": "Необраб.",
    "sendingRequest": "Отправка запроса..."
  },
  "Table": {
    "noHeaders": "Нет заголовков...",
    "key": "Ключ",
    "value": "Значение"
  },
  "NotFound": {
    "notFound": "Страница не найдена. Вы можете перейти на",
    "mainPage": "Главную\u00a0страницу"
  },
  "History": {
    "title": "История запросов",
    "description": "Вы не выполнили ни одного запроса. Здесь пусто. Попробуйте:"
  },
  "Graphiql": {
    "title": "GraphiQL клиент",
    "send": "Отправить",
    "set": "Установить",
    "endpointURL": "URL конечной точки",
    "copyTextSuccess": "Текст скопирован в буфер обмена",
    "copyTextError": "Ошибка: текст не скопирован в буфер обмена",
    "headers": "Заголовки",
    "variables": "Переменные",
    "key": "Ключ",
    "value": "Значение"
  }
}
