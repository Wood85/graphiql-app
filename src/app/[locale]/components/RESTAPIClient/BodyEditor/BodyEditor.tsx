'use client';

import { useTranslations } from 'next-intl';
import { useState } from 'react';

import Editor from '@monaco-editor/react';

import Button from '@/components/UI/Button/Button';
import { replaceInHistory } from '@/utils/replaceHistory';
import HeadersEditor from '../HeadersEditor/HeadersEditor';
import VariablesEditor from '../VariablesEditor/VariablesEditor';

import styles from './BodyEditor.module.scss';

interface IProps {
  setBody: React.Dispatch<React.SetStateAction<string>>;
  body: string;
}

type TSelect = 'headers' | 'body' | 'variables';

type TFormat = 'json' | 'text';

export const dynamic = 'force-dynamic';

function BodyEditor({ setBody, body }: IProps): JSX.Element {
  const t = useTranslations('Restapi');

  const [select, setSelect] = useState<TSelect>('headers');
  const [editorValue, setEditorValue] = useState<string>('{}');
  const [format, setFormat] = useState<TFormat>('json');

  function handleEditorChange(value: string | undefined): void {
    if (value !== undefined) {
      setBody(value);
      setEditorValue(value);
    }
  }
  return (
    <div className={styles.container}>
      <div className={styles.switchers_container}>
        <div className={styles.switcher}>
          <Button
            className={`${styles.button} ${select === 'headers' ? styles.selected : ''}`}
            onClick={() => {
              setSelect('headers');
            }}
          >
            {t('headers')}
          </Button>
          <Button
            className={`${styles.button} ${select === 'body' ? styles.selected : ''}`}
            onClick={() => {
              setSelect('body');
            }}
          >
            {t('body')}
          </Button>
          <Button
            className={`${styles.button} ${select === 'variables' ? styles.selected : ''}`}
            onClick={() => {
              setSelect('variables');
            }}
          >
            {t('variables')}
          </Button>
        </div>
      </div>
      {select === 'headers' && <HeadersEditor />}
      {select === 'body' && (
        <div
          className={styles.editor_container}
          onBlur={() => {
            replaceInHistory('body', editorValue);
          }}
        >
          <div className={styles.format_switcher}>
            <Button
              className={`${styles.format_button} ${format === 'json' ? styles.selected_format : ''}`}
              onClick={() => {
                setFormat('json');
              }}
            >
              JSON
            </Button>
            <Button
              className={`${styles.format_button} ${format === 'text' ? styles.selected_format : ''}`}
              onClick={() => {
                setFormat('text');
              }}
            >
              {t('text')}
            </Button>
          </div>
          <Editor
            height='280px'
            defaultValue='{}'
            language={format}
            theme='vs-light'
            value={body ?? editorValue}
            options={{
              detectIndentation: false,
              fontFamily: '"Cera Pro"',
              fontSize: 16,
              minimap: { enabled: false },
              padding: { top: 42, bottom: 5 },
              renderLineHighlight: 'none',
              scrollBeyondLastLine: false,
              tabSize: 2,
              wordWrap: 'on',
              wrappingIndent: 'deepIndent',
              wrappingStrategy: 'advanced',

              inlineSuggest: {
                enabled: true,
                showToolbar: 'onHover',
                mode: 'subword',
                suppressSuggestions: false,
              },
              autoIndent: 'advanced',
              formatOnPaste: true,
              formatOnType: true,
              autoClosingBrackets: 'always',
            }}
            className={styles.editor}
            onChange={handleEditorChange}
          />
        </div>
      )}
      {select === 'variables' && <VariablesEditor />}
    </div>
  );
}

export { BodyEditor };
