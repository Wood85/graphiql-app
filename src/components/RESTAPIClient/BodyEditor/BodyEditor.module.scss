@import './../../../styles/variables.scss';

.switchers_container {
  display: flex;
  // justify-content: space-between;
}

.button:focus-visible {
  border: 1px solid $dark-blue;
}

.switcher {
  display: flex;
  gap: 5px;
}

.editor_container {
  position: relative;

  height: 280px;
  width: 100%;
}

.format_switcher {
  position: absolute;
  top: 10px;
  left: 10px;
  z-index: 10;

  display: flex;
  align-items: center;
  margin-right: 6px;

  font-size: 14px;

  @media screen and (max-width: 500px) {
    margin-right: 0;
  }
}

.format_button {
  padding: 0 8px;
  border: 1px solid transparent;
  height: 25px;

  color: $dark-blue;
  font-size: 16px;

  background-color: $gray;

  transition: all 0.3s ease-in;

  @media screen and (max-width: 340px) {
    padding: 0 3px;
  }

  &:hover {
    transition: all 0.3s ease-in;
  }

  &:first-child {
    border-top-right-radius: 0;
    border-bottom-right-radius: 0;
  }

  &:last-child {
    border-top-left-radius: 0;
    border-bottom-left-radius: 0;
  }

  &:focus-visible {
    border: 1px solid $dark-blue;
  }
}

.format_button.selected_format,
.format_button.selected_format:hover {
  color: $white;

  background-color: $blue;
  cursor: default;

  transition: all 0.3s ease-in;
}

.button {
  background-color: $white;

  color: $blue;
  font-size: 20px;

  transition: all 0.3s ease-in;

  @media screen and (max-width: 500px) {
    padding: 0;

    font-size: 16px;
  }

  &:not(:disabled):hover {
    color: $dark-blue;

    background-color: $white;

    transition: all 0.3s ease-in;
  }
}

.selected {
  color: $dark-blue;
}

.editor {
  width: 100%;
  border: 1px solid $blue;

  font-size: 20px;
}
