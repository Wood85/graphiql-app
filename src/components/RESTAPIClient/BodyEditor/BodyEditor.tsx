'use client';

import { useState } from 'react';
import Editor from '@monaco-editor/react';
import Button from '@/components/UI/Button/Button';
import styles from './BodyEditor.module.scss';
import { TableEditor } from '../TableEditor/TableEditor';

interface IProps {
  body: string;
  setBody: React.Dispatch<React.SetStateAction<string>>;
}

type TSelect = 'headers' | 'body' | 'variables';

function BodyEditor({ body, setBody }: IProps): JSX.Element {
  const [select, setSelect] = useState<TSelect>('headers');

  function handleEditorChange(value: string | undefined): void {
    console.log('here is the current model value:', value);
  }
  return (
    <div className={styles.container}>
      <div className={styles.switcher}>
        <Button
          className={`${styles.button} ${select === 'headers' ? styles.selected : ''}`}
          onClick={() => {
            setSelect('headers');
          }}
        >
          Headers
        </Button>
        <Button
          className={`${styles.button} ${select === 'body' ? styles.selected : ''}`}
          onClick={() => {
            setSelect('body');
          }}
        >
          Body
        </Button>
        <Button
          className={`${styles.button} ${select === 'variables' ? styles.selected : ''}`}
          onClick={() => {
            setSelect('variables');
          }}
        >
          Variables
        </Button>
      </div>
      {select === 'headers' && <TableEditor />}
      {select === 'body' && (
        <Editor
          height='280px'
          defaultValue='{}'
          defaultLanguage='javascript'
          theme='vs-light'
          options={{
            inlineSuggest: {
              enabled: true,
              showToolbar: 'onHover',
              mode: 'subword',
              suppressSuggestions: false,
            },
            fontSize: 18,
            formatOnType: true,
            autoClosingBrackets: 'always',
            minimap: { enabled: false },
          }}
          className={styles.editor}
          onChange={handleEditorChange}
        />
        // 	<textarea
        //   className={styles.editor}
        //   placeholder='Request Body (JSON)'
        //   value={body}
        //   rows={10}
        //   onChange={(e) => {
        //     setBody(e.target.value);
        //   }}
        // />
      )}
    </div>
  );
}

export { BodyEditor };
