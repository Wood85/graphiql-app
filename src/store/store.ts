import { configureStore } from '@reduxjs/toolkit';
import restFullReducer from '@/store/reducers/restFullSlice';

export const store = configureStore({
  reducer: {
    rest: restFullReducer,
  },
});

export default store;

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
