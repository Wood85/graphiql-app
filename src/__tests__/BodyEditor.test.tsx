import { BodyEditor } from '@/app/[locale]/components/RESTAPIClient/BodyEditor/BodyEditor';
import { store } from '@/store/store';
import { fireEvent, render } from '@testing-library/react';
import { NextIntlClientProvider } from 'next-intl';
import { Provider } from 'react-redux';
import { describe, expect, it } from 'vitest';
import messages from '../../messages/en.json';

const locale = 'en';

const ONE = 1;
describe('BodyEditor component', () => {
  it('renders correctly', () => {
    const { container } = render(
      <Provider store={store}>
        <NextIntlClientProvider messages={messages} locale={locale}>
          <BodyEditor setBody={() => {}} />
        </NextIntlClientProvider>
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });

  it('renders headers editor by default', () => {
    const { getByText } = render(
      <Provider store={store}>
        <NextIntlClientProvider messages={messages} locale={locale}>
          <BodyEditor setBody={() => {}} />
        </NextIntlClientProvider>
      </Provider>,
    );
    expect(getByText('Headers')).toBeDefined();
  });

  it('switches to body editor', () => {
    const { getByText, getByRole } = render(
      <Provider store={store}>
        <NextIntlClientProvider messages={messages} locale={locale}>
          <BodyEditor setBody={() => {}} />
        </NextIntlClientProvider>
      </Provider>,
    );
    const bodyButton = getByRole('button', { name: 'Body' });
    fireEvent.click(bodyButton);
    expect(getByText('Body')).toBeDefined();
  });

  it('switches to variables editor', () => {
    const { getByText, getByRole } = render(
      <Provider store={store}>
        <NextIntlClientProvider messages={messages} locale={locale}>
          <BodyEditor setBody={() => {}} />
        </NextIntlClientProvider>
      </Provider>,
    );
    const variablesButton = getByRole('button', { name: 'Variables' });
    fireEvent.click(variablesButton);
    expect(getByText('Variables')).toBeDefined();
  });
});
