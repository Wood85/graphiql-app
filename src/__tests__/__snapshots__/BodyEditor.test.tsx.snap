// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BodyEditor component > renders correctly 1`] = `
<div>
  <div
    class="_container_625087"
  >
    <div
      class="_switchers_container_625087"
    >
      <div
        class="_switcher_625087"
      >
        <button
          class="_button_e7dbe0 _button_625087 _selected_625087"
          type="button"
        >
          Headers
        </button>
        <button
          class="_button_e7dbe0 _button_625087 "
          type="button"
        >
          Body
        </button>
        <button
          class="_button_e7dbe0 _button_625087 "
          type="button"
        >
          Variables
        </button>
      </div>
    </div>
    <div
      class="_container_30778c"
    >
      <table
        class="_table_30778c"
        role="grid"
      >
        <thead>
          <tr>
            <td
              class="_td_30778c _td_1_30778c"
            />
            <td
              class="_td_30778c _td_2_30778c _td_title_30778c"
            >
              Key
            </td>
            <td
              class="_td_30778c _td_3_30778c _td_title_30778c"
            >
              Value
            </td>
            <td
              class="_td_30778c _td_4_30778c _td_title_30778c"
            />
          </tr>
        </thead>
        <tbody>
          <tr>
            <td
              class="_td_8562a9 _td_1_8562a9"
            >
              <label>
                <input
                  class="_checkbox_hide_94586e"
                  data-testid="checkbox"
                  type="checkbox"
                />
                <div
                  class="_checkbox_94586e _checkbox_active_94586e"
                />
              </label>
            </td>
            <td
              class="_td_8562a9 _td_2_8562a9 false"
            >
              <label
                for="Accept"
              >
                Accept
              </label>
            </td>
            <td
              class="_td_8562a9 _td_3_8562a9 false"
            >
              <label
                for="Accept"
              >
                */*
              </label>
            </td>
            <td
              class="_td_8562a9 _td_4_8562a9"
            />
          </tr>
          <tr>
            <td
              class="_td_8562a9 _td_1_8562a9"
            >
              <label>
                <input
                  class="_checkbox_hide_94586e"
                  data-testid="checkbox"
                  type="checkbox"
                />
                <div
                  class="_checkbox_94586e _checkbox_active_94586e"
                />
              </label>
            </td>
            <td
              class="_td_8562a9 _td_2_8562a9 false"
            >
              <label
                for="Accept-Encoding"
              >
                Accept-Encoding
              </label>
            </td>
            <td
              class="_td_8562a9 _td_3_8562a9 false"
            >
              <label
                for="Accept-Encoding"
              >
                gzip, deflate, br
              </label>
            </td>
            <td
              class="_td_8562a9 _td_4_8562a9"
            />
          </tr>
          <tr>
            <td
              class="_td_8562a9 _td_1_8562a9"
            >
              <label>
                <input
                  class="_checkbox_hide_94586e"
                  data-testid="checkbox"
                  type="checkbox"
                />
                <div
                  class="_checkbox_94586e _checkbox_active_94586e"
                />
              </label>
            </td>
            <td
              class="_td_8562a9 _td_2_8562a9 false"
            >
              <label
                for="Connection"
              >
                Connection
              </label>
            </td>
            <td
              class="_td_8562a9 _td_3_8562a9 false"
            >
              <label
                for="Connection"
              >
                keep-alive
              </label>
            </td>
            <td
              class="_td_8562a9 _td_4_8562a9"
            />
          </tr>
        </tbody>
        <tfoot>
          <tr
            data-testid="row_editor"
          >
            <td
              aria-label="checkbox"
              class="_td_b1edd3 _td_1_b1edd3"
              role="gridcell"
            />
            <td
              aria-label="key"
              class="_td_b1edd3 _td_2_b1edd3"
              role="gridcell"
            />
            <td
              aria-label="value"
              class="_td_b1edd3 _td_3_b1edd3"
              role="gridcell"
            />
            <td
              aria-label="btns"
              class="_td_b1edd3 _td_4_b1edd3"
              role="gridcell"
            />
          </tr>
        </tfoot>
      </table>
    </div>
  </div>
</div>
`;
